components/expenses/Expense.js

```diff
@@ -500,8 +505,50 @@ class Expense extends React.Component {
           {this.state.error && (
             <MessageBox type="error" withIcon mb={2}>
               {formatErrorMessage(intl, this.state.error)}
+              <br />
+              {formatErrorMessage(intl, this.state.error).startsWith(
+                'Host has two-factor authentication enabled for large payouts.',
+              ) && (
+                <FormattedMessage
+                  id="PayExpenseModal.HostTwoFactorAuthEnabled"
+                  defaultMessage="Please go to your {settings} to enable two-factor authentication on your account."
+                  values={{
+                    settings: (
+                      <StyledLink as={Link} route="editCollective" params={{ slug: LoggedInUser.collective.slug }}>
+                        account settings
+                      </StyledLink>
+                    ),
+                  }}
+                />
+              )}
             </MessageBox>
           )}
+          {this.state.error && formatErrorMessage(intl, this.state.error).startsWith('Two-factor authentication') && (
```

Comment:
Same here, we shouldn't detect the error type from the result of `formatErrorMessage`